### PR TITLE
MODULES-9729 update exit code for nil cases

### PR DESCRIPTION
**Pull request by sheenaajay**
Request for a review

### CODE REVIEW

File: lib/puppet_litmus/serverspec.rb [hunk 1]
```diff
@@ -184,7 +184,11 @@ def run_bolt_task(task_name, params = {}, opts = {})
     else
       raise "task failed\n`#{task_name}`\n======\n#{result}" if opts[:expect_failures] != true
 
-      result_obj[:exit_code] = result.first['result']['_error']['details'].fetch('exitcode', 255)
```
Comment (michaeltlombardi) [inline lib/puppet_litmus/serverspec.rb]: My understanding was that this line should return 255 if nil, or the exit code if it exists. 🤔

Comment (michaeltlombardi) [reply lib/puppet_litmus/serverspec.rb]: Do you have an example of where this has failed in an existing test?

Comment (sheenaajay) [reply lib/puppet_litmus/serverspec.rb]: Sure @michaeltlombardi .This is the example where its failing and throws nil error. Result set doesnt have[ '_error']['details']

Fragment of testcase
params = { 'fact' => '--foo' }
result = run_bolt_task('facter_task', params, expect_failures: true)

Results
=> [{"node"=>"mhyo0hxupa8ma56.delivery.puppetlabs.net",
  "target"=>"mhyo0hxupa8ma56.delivery.puppetlabs.net",
  "action"=>"task",
  "object"=>"facter_task",
  "status"=>"failure",
  "result"=>{"_error"=>{"kind"=>"facter_task/failure", "msg"=>"Exit 1 running /opt/puppetlabs/puppet/bin/facter -p --json --foo: error: unrecognised option '--foo'\n"}}}]

Comment (michaeltlombardi) [reply lib/puppet_litmus/serverspec.rb]: Ahhhhhhhhhh. That makes sense! I didn't know we could return an error without details. :sob:

**Pull request merged**